client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/launchpad-site-preview.tsx

```diff
@@ -14,25 +22,65 @@ const LaunchpadSitePreview = ( {
 	siteSlug: string | null;
 	flow: string | null;
 } ) => {
+	const site = useSite();
 	const translate = useTranslate();
 	const { globalStylesInUse, shouldLimitGlobalStyles } = usePremiumGlobalStyles();
 
 	const previewUrl = siteSlug ? 'https://' + siteSlug : null;
 	const devicesToShow: Device[] = [ DEVICE_TYPES.COMPUTER, DEVICE_TYPES.PHONE ];
 	let defaultDevice = flow === NEWSLETTER_FLOW ? DEVICE_TYPES.COMPUTER : DEVICE_TYPES.PHONE;
 	const isVideoPressFlow = VIDEOPRESS_FLOW === flow;
+	const isBusinessPlan = site?.plan?.product_slug
+		? isWpComBusinessPlan( site?.plan?.product_slug )
+		: false;
+	const isEcommercePlan = site?.plan?.product_slug
+		? isWpComEcommercePlan( site?.plan?.product_slug )
+		: false;
 
 	if ( isVideoPressFlow ) {
 		const windowWidth = window.innerWidth;
 		defaultDevice = windowWidth >= 1000 ? DEVICE_TYPES.COMPUTER : DEVICE_TYPES.PHONE;
 	}
 
+	const usePreviewSiteLinksQueryEnabled =
```

Comment:
This code looks fine as is. But there's a lot going on here. 

We're adding about 50 lines of code to (basically doubling) this launchpad-site-prevew.tsx file, all to provide support for this one specialized use case of preview links. 

Ultimately it looks like we use three items from all this extra code to influence the output in the formatPreviewUrl method: shareCode, isPreviewLinksLoading, isCreatingSitePreviewLinks

This is really a matter of taste, but I wonder if it might be cleaner to put all this logic in our own hook, maybe even just inside the launchpad folder. Call it useShareCode. Almost all of this extra code would go there. 

Then in this file, we could just import that hook and do something like the following to get the elements we need in this file. 

`const { shareCode, isPreviewLinksLoading, isCreatingSitePreviewLinks } = useShareCode( site?.ID );`

I did try to figure out if there was some way to simplify all the logic and calls here too, but doesn't seem like it too much. So the custom hook just encapsulates all this extra share code login in one place. 

I don't feel too strongly on this, so if we prefer to leave it here, thats fine too.

Reply:
This is a great suggestion. There is a lot of code being added here. I prefer to separate it out into its own new hook. This also helps make it reusable (I think there are other areas of Calypso that could use this solution for the "coming soon" issue.